modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -353,7 +353,11 @@ private var credentialStore = SynchronizedDictionary<Int, URLCredential>()
         default:
 
             guard let data = data, !data.isEmpty else {
-                completion(.failure(ErrorResponse.error(httpResponse.statusCode, nil, response, DecodableRequestBuilderError.emptyDataResponse)))
+                if T.self is ExpressibleByNilLiteral.Type {
+                    completion(.success(Response(response: httpResponse, body: Optional<T>.none as! T)))
+                } else {
+                    completion(.failure(ErrorResponse.error(httpResponse.statusCode, nil, response, DecodableRequestBuilderError.emptyDataResponse)))
+                }
```

Comment:
@Paul-Etienne Hey, could you please make the same change to Alamofire please? Thanks

Reply:
Done. I wasn't able to test it though, but it should work as well as the URLSession one, you let me know what you think 👍

Reply:
Can this create a double optional?
In case the response is already optional?

Reply:
From my testing, this is effectively creating a nil response body of type T? when the response type is an optional, and not a double optional.

Reply:
Thanks 👍